services/src/main/java/org/keycloak/services/resources/admin/RealmAdminResource.java

```diff
@@ -588,10 +588,17 @@ public void deleteSession(@PathParam("session") String sessionId) {
         auth.users().requireManage();
 
         UserSessionModel userSession = session.sessions().getUserSession(realm, sessionId);
-        if (userSession == null) throw new NotFoundException("Sesssion not found");
-        AuthenticationManager.backchannelLogout(session, realm, userSession, uriInfo, connection, headers, true);
-        adminEvent.operation(OperationType.DELETE).resource(ResourceType.USER_SESSION).resourcePath(uriInfo).success();
+        UserSessionModel offlineSession = session.sessions().getOfflineUserSession(realm, sessionId);
```

Comment:
Immediate searching for offline session can have a negative impact on performance since the session can be stored on another node. The offline session should only be looked up when `userSession == null`.

Reply:
The key thing is that extra network roundtrip should be prevented in clustered environments.

I dug a bit more into the code. If offline session is invalidated, online is invalidated too, not vice versa. Hence the code needs first to lookup offline session, and only if that is not available look up online session.

If you update the code that way, it can be merged.